runtime/src/iree/runtime/demo/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ iree_cc_binary(
     iree::runtime::unified
 )
 
-if(NOT IREE_HAL_EXECUTABLE_LOADER_VMVX_MODULE OR NOT IREE_TARGET_BACKEND_VMVX)
+if(NOT IREE_BUILD_TESTS OR NOT IREE_HAL_EXECUTABLE_LOADER_VMVX_MODULE OR NOT IREE_TARGET_BACKEND_VMVX)
```

Comment:
this isn't technically a test dir but down below there's a dep on a `iree::runtime::testdata::simple_mul_module_c`

Reply:
These aren't really tests and I'd not want to disable them based on that - they're binaries that ensure the system builds correctly. Maybe this should be moved to samples/, but since it's a strong indicator of a successful build (vs a unit/integration test) I'm not sure... I'll let @ScottTodd decide.

Reply:
Looks like the "testdata" folder there is leftover from style on previous projects where the line between "sample" and "test" was blurrier. In IREE now we have `IREE_BUILD_TESTS` and `IREE_BUILD_SAMPLES` that are separable.

Could rename `iree::runtime::testdata` to `iree::runtime::demodata` by renaming the directory or (my preference) just collapse that 'testdata' directory into the `demo` folder.

```
iree
  runtime
    demo
      hello_world_terse.c
      simple_mul.mlir
      CMakeLists.txt
```

then I'd also update the path in this comment (didn't find other uses but should confirm): https://github.com/iree-org/iree/blob/63a2d141adf7b3911bc94c7f779e1896fb25aecb/tools/iree-dump-instruments-main.c#L268